thanos/charts/templates/ruler.yaml

```
@@ -0,0 +1,87 @@
+# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+# SPDX-License-Identifier: Apache-2.0
+
+{{ if .Values.ruler.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    {{- include "plugin.labels" . | nindent 4 }}
+    {{- include "thanos.labels" . | nindent 4 }}
+    {{- if .Values.thanos.ruler.deploymentLabels }}
+    {{ toYaml .Values.thanos.ruler.deploymentLabels | nindent 4 }}
+    {{- end }}
+  name: {{ include "release.name" . }}-ruler
+
+spec:
+  image: "{{ .Values.thanos.image.repository }}:{{ .Values.thanos.image.tag | default .Chart.AppVersion }}"
+  paused: {{ $.Values.ruler.paused }}
+  logLevel: {{ default "info" $.Values.ruler.logLevel }}
+  {{- if $.Values.ruler.retention }}
+  retention: {{ $.Values.ruler.retention | quote }}
+  {{- end }}
+  {{- if $.Values.rbac.create }}
+  serviceAccountName: {{ $.Values.serviceAccount.name }}
```

Would go with a default here: 
```suggestion
  serviceAccountName: {{ include "release.name" . }}-ruler
```